posts/2021-06-16-graykull-step-by-step.rst

```diff
@@ -0,0 +1,35 @@
+.. post:: 2021-06-16
+   :tags: Grayskull
+   :author: ForgottenProgramme
+   :redirect: 2021/06/16/grayskull-step-by-step
+
+Contributing Packages To conda-forge Using Grayskull
+====================================================
+
+When contributing packages to conda-forge, Grayskull can make your life much easier.
+Grayskull generates recipes for Python packages hosted on PyPI.
+
+As the `introduction <https://github.com/conda-incubator/grayskull#introduction>`__ for Grayskull reads; “The main goal of this project is to generate concise recipes for conda-forge.”
+In this tutorial we learn how to contribute a Python package to the conda-forge channel using Grayskull to generate the recipe.
+
+Let us get started.
+
+1. Install ``grayskull`` using ``conda`` through the ``conda-forge`` channel:
+    ``conda install -c conda-forge grayskull``
+2. Fork and clone the conda-forge `staged-recipes repository<https://github.com/conda-forge/staged-recipes>`__ from GitHub.
+3. Fetch a new branch from the ``master branch``.
+4. Through CLI enter inside the ‘staged-recipes’ directory.
+5. Call ``grayskull`` and pass the ``pypi`` repository, followed by the name of the package you want to contribute to conda-forge. For example:
+   ``grayskull pypi abc``
```

Comment:
```suggestion
   ``grayskull pypi abc --strict-conda-forge``
```

I would recommend adding this option when generating the recipes specific for the ``conda-forge`` channel.

Reply:
It would be nice to explain that option in the post.

Reply:
@marcelotrevisani 
> I would recommend adding this option when generating the recipes specific for the `conda-forge` channel.

Would you please explain ``grayskull pypi abc --strict-conda-forge`` to me?
Does it mean: get the 'abc' package from PyPI and through the conda-forge channel only ?

Reply:
Hey, no no, it is different

When you use that option it will remove some selectors which are not necessary for conda-forge and it will try to adapt some recipes to fit better in the conda-forge ecosystem.

For instance, the recipe generated for ``pytest`` ``5.3.1`` without the ``--strict-conda-forge`` will be something like:

```yaml
{% set name = "pytest" %}
{% set version = "5.3.1" %}


package:
  name: {{ name|lower }}
  version: {{ version }}

source:
  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/pytest-{{ version }}.tar.gz
  sha256: f67403f33b2b1d25a6756184077394167fe5e2f9d8bdaab30707d19ccec35427

build:
  number: 0
  skip: true   # [py2k]
  entry_points:
    - pytest=pytest:main
    - py.test=pytest:main
  script: {{ PYTHON }} -m pip install . -vv

requirements:
  host:
    - pip
    - python
    - setuptools >=40.0
    - setuptools_scm
  run:
    - atomicwrites >=1.0  # [win]
    - attrs >=17.4.0
    - colorama  # [win]
    - importlib-metadata >=0.12  # [py<38]
    - more-itertools >=4.0.0
    - packaging
    - pathlib2 >=2.2.0  # [py<36]
    - pluggy >=0.12,<1.0
    - py >=1.5.0
    - python
    - wcwidth
```

with ``--strict-conda-forge`` it will drop that selector in ``skip: True  # [py2k]`` and also removed the ``pathlib  # [py<36]``

```yaml
{% set name = "pytest" %}
{% set version = "5.3.1" %}


package:
  name: {{ name|lower }}
  version: {{ version }}

source:
  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/pytest-{{ version }}.tar.gz
  sha256: f67403f33b2b1d25a6756184077394167fe5e2f9d8bdaab30707d19ccec35427

build:
  number: 0
  entry_points:
    - pytest=pytest:main
    - py.test=pytest:main
  script: {{ PYTHON }} -m pip install . -vv

requirements:
  host:
    - pip
    - python
    - setuptools >=40.0
    - setuptools-scm
  run:
    - atomicwrites >=1.0  # [win]
    - attrs >=17.4.0
    - colorama  # [win]
    - importlib-metadata >=0.12  # [py<38]
    - more-itertools >=4.0.0
    - packaging
    - pluggy >=0.12,<1.0
    - py >=1.5.0
    - python
    - wcwidth
```

Reply:
ohhh, I see. 
But this blog is meant for new users and maybe all this information about using ``--strict-conda-forge`` might seem intimidating at first. Maybe we can skip it for this blog. What do you think? 
Have you explained this in any of your blogs on grayskull? 

If yes, we could do something like this:

5. Call ``grayskull`` and pass the ``pypi`` repository, followed by the name of the package you want to contribute to conda-forge. For example:
``grayskull pypi abc``
To better fit your recipe to the conda-forge ecosystem, you could instead give the following command: 
``grayskull pypi abc --strict-conda-forge``
To know more read this (a link to your blog).

Reply:
@marcelotrevisani 
done.